### PR TITLE
Adding Filter to list dialog

### PR DESCRIPTION
**Pull request by h-haddad**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt [hunk 1]
```diff
@@ -46,6 +52,29 @@ class ListTable(val model: VListDialog) : Grid<List<Any?>>() {
         .setKey(col.toString())
     }
   }
+  /**
+   * Install filters on all properties.
+   */
+  fun passesFilter(model: VListDialog?) {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt:58]: `installFilters` a better name for this method

File: galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt [hunk 2]
```diff
@@ -48,14 +48,13 @@ class ListTable(val model: VListDialog) : Grid<List<Any?>>() {
       addColumn {
         formatObject(it[col], col)
       }.setHeader(model.getColumnName(col))
-        .setAutoWidth(true)
-        .setKey(col.toString())
+              .setAutoWidth(true).key = col.toString()
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt:51]: for better readability i'd prefer to keep it with `setKey`

File: galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt [hunk 3]
```diff
@@ -42,9 +48,31 @@ class ListTable(val model: VListDialog) : Grid<List<Any?>>() {
       addColumn {
         formatObject(it[col], col)
       }.setHeader(model.getColumnName(col))
-        .setAutoWidth(true)
-        .setKey(col.toString())
+              .setAutoWidth(true).setKey(col.toString())
```
Comment (SNsir) [inline galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt]: reformat

File: galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt [hunk 4]
```diff
@@ -42,9 +48,31 @@ class ListTable(val model: VListDialog) : Grid<List<Any?>>() {
...
+    }
+  }
```
Comment (SNsir) [inline galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/list/ListTable.kt:54]: add empty line

**Review by SNsir — APPROVED** [review]

**Pull request merged**